runtime/compiler/trj9/optimizer/JProfilingValue.cpp

```diff
@@ -280,15 +280,42 @@ TR_JProfilingValue::lowerCalls()
 
          dumpOptDetails(comp(), "%s Replacing profiling placeholder n%dn with value profiling trees\n",
             optDetailString(), child->getGlobalIndex());
-
+         TR::TreeTop *prevTT = cursor->getPrevTreeTop();
+         TR::TreeTop *nextTT = cursor->getNextTreeTop();
+         TR::Node *prevTTNode = prevTT->getNode();
          // Extract the arguments and add the profiling trees
          TR::Node *value = child->getFirstChild();
          TR_AbstractHashTableProfilerInfo *table = (TR_AbstractHashTableProfilerInfo*) child->getSecondChild()->getAddress();
+         // While lowering value profiling calls, Scan the previous trees with the next trees to make sure by lowering the value profiler calls,
+         // We are not splitting blocks which orphans compressedref anchor nodes.
```

Comment:
I'm not clear on why prevTT is the right node to look at - the node we care about is the first child of the call that is going to be replaced correct? The treetop for the call is cursor not cursor->getPrevTreeTop()